### ISSUE TITLE
Fallback `ObjectModel.attr___new__` has signature `(self, cls)`, so `self.__new__(type(self))` / `super().__new__(cls)` / C-type `__new__` calls report missing or unexpected arguments

### ISSUE THREAD

**Issue by ekanshul** [pylint-dev/astroid#3237]
### Steps to reproduce

`ObjectModel.attr___new__` is the fallback `__new__` returned whenever the real one is not found: `instance.__new__`, a class with an ancestor that cannot be inferred, and (through `raw_building`'s `ImportFrom("builtins", "__new__")`) the `__new__` of every C type outside `builtins` that defines its own `tp_new`. It is built from

```python
def __new__(self, cls): return cls()
```

but `BoundMethod.implicit_parameters()` is `0` for `__new__` (the class is passed explicitly), so the class argument lands in `self` and `cls` is reported as missing, while any further argument is "unexpected":

```python
import astroid

nodes = astroid.extract_node("""
class A:
    def clone(self):
        self.__new__(type(self))  #@

from missing import Base

class B(Base):
    def __new__(cls, value):
        super().__new__(cls)  #@

import socket
socket.socket.__new__(socket.socket, family=1)  #@
""")
for node in nodes:
    func = next(node.func.infer())
    print(func, func.argnames(), func.implicit_parameters())
```

```
<BoundMethod __new__ of builtins.object at ...> ['self', 'cls'] 0
<BoundMethod __new__ of builtins.object at ...> ['self', 'cls'] 0
<BoundMethod __new__ of builtins.object at ...> ['self', 'cls'] 0
```

pylint turns each of these into a false positive:

```
no-value-for-parameter: No value for argument 'cls' in classmethod call      (the first two)
unexpected-keyword-arg: Unexpected keyword argument 'family' in classmethod call
no-value-for-parameter: No value for argument 'cls' in classmethod call      (the socket call)
```

Real-world hits: `self.__new__(type(self))` (pylint-dev/pylint#8325), `super().__new__(cls)` in a class whose base comes from an uninstalled package (the comment on that issue), `cls.__new__(cls, **kwargs)` in the stdlib's `ssl.SSLSocket._create`, `array.array.__new__(cls, "b", data)`, and on Python 3.11 `datetime.datetime.__new__` (pylint-dev/pylint#10670, which only stopped reproducing on 3.12+ because `datetime` now resolves to `_pydatetime`).

### Current behavior

The fallback `__new__` has the signature `(self, cls)`.

### Expected behavior

The first parameter should be the class, and the rest should accept anything, like the sibling `attr___init__` fallback already does (`def __init__(self, *args, **kwargs)`, with the comment "necessary not to trigger warnings about missing or extra parameters for `__init__` methods we don't infer correctly"). Something like `def __new__(cls, *args, **kwargs): return cls()`; inference of the call result is unaffected since it goes through `UnboundMethod._infer_builtin_new`.

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output

4.4.0-dev0 (main, 94b7090f); also on 4.0.4 / 4.2.0b5.
